contracts/RFQ.sol

```diff
@@ -80,13 +81,23 @@ contract RFQ is IRFQ, StrategyBase, ReentrancyGuard, SignatureValidator, BaseLib
      *************************************************************/
     function fill(
         RFQLibEIP712.Order calldata _order,
+        SpenderLibEIP712.SpendWithPermit calldata _spendMakerAssetToRFQ,
+        SpenderLibEIP712.SpendWithPermit calldata _spendTakerAssetToRFQ,
         bytes calldata _mmSignature,
-        bytes calldata _userSignature
+        bytes calldata _userSignature,
+        bytes calldata _spendMakerAssetToRFQSig,
+        bytes calldata _spendTakerAssetToRFQSig
     ) external payable override nonReentrant onlyUserProxy returns (uint256) {
         // check the order deadline and fee factor
         require(_order.deadline >= block.timestamp, "RFQ: expired order");
         require(_order.feeFactor < LibConstant.BPS_MAX, "RFQ: invalid fee factor");
 
+        // check the spender deadline and RFQ address
+        require(_spendMakerAssetToRFQ.expiry >= block.timestamp, "RFQ: expired maker spender");
+        require(_spendMakerAssetToRFQ.requester == address(this), "RFQ: invalid RFQ address");
+        require(_spendTakerAssetToRFQ.expiry >= block.timestamp, "RFQ: expired taker spender");
+        require(_spendTakerAssetToRFQ.requester == address(this), "RFQ: invalid RFQ address");
```

Comment:
I think these checks could be enforced by Spender

Reply:
Yes, you're right!
In fact, the expired check is already in the spender.spendFromUserToWithPermit() func.
But as I understand it, the requester checks need to be kept in each strategy contract to ensure the user is using the correct strategy contract.

Reply:
But if we could check `msg.sender` to be the `requester` in Spender like Charles's suggestion, then I think we could skip requester check in strategy contracts.

Reply:
Yes, in the next version I will put this check into the `spender.spendFromUserToWithPermit()` and change the check address from address(this) to msg.sender ( = the strategy contract) :)